src/aleph/vm/controllers/qemu/cloudinit.py

```diff
@@ -0,0 +1,138 @@
+import base64
+import json
+from pathlib import Path
+from tempfile import NamedTemporaryFile
+from typing import Union
+
+import yaml
+from aleph_message.models import ItemHash
+
+from aleph.vm.conf import settings
+from aleph.vm.controllers.interface import AlephControllerInterface
+from aleph.vm.hypervisors.firecracker.config import Drive
+from aleph.vm.utils import is_command_available, run_in_subprocess
+
+# https://cloudinit.readthedocs.io/en/latest/reference/datasources/nocloud.html
```

Comment:
How is this related to the code? Maybe you can add a bit of explanation (for n00bs like me)

Reply:
Keep the link there for easy reference, but yeah we should have a docstring for the whole module, will add it